### PR TITLE
fixed bugs

### PR DESCRIPTION
**Pull request by dkok97**
fixed issue #39 and #37 .
I don't think issue #38 needs to be fixed, the email is updated once you click out of the text space.

### CODE REVIEW

File: src/components/pages/profile/mentorship.js [hunk 1]
```diff
@@ -83,15 +83,26 @@ class Mentorship extends React.Component {
         }
     }
 
+	componentWillMount() {
```
Comment (ramsgoli) [inline src/components/pages/profile/mentorship.js]: this should all be done in componentDidMount() i believe. Calling this.setState in componentWillMount doesnt always trigger a rerender. See https://reactjs.org/docs/react-component.html#componentwillmount
Also,
`let newcourses = this.props.mentor.courses.map(obj => obj.name);`
is fine

File: src/components/pages/profile/mentorship.js [hunk 2]
```diff
@@ -83,15 +83,26 @@ class Mentorship extends React.Component {
...
+		}
+	}
+
     componentWillReceiveProps(nextProps) {
         this.setState({
 			bio: nextProps.mentor.bio,
-			courses: nextProps.mentor.courses.map(function(obj) {
-				return obj.name
-			}),
+			courses: nextProps.mentor.courses.map(obj=>{return obj.name})
```
Comment (ramsgoli) [inline src/components/pages/profile/mentorship.js]: `courses: nextProps.mentor.courses.map(obj => obj.name)`

File: src/components/pages/profile/mentorship.js (hunk 1 above)
Comment (dkok97) [reply src/components/pages/profile/mentorship.js]: I first had it in componentDidMount but it went into an infinite loop

Comment (dkok97) [reply src/components/pages/profile/mentorship.js]: ok nvm i changed it to componentDidMount

**Pull request merged**